src/haven/instrument/robot.py

```diff
@@ -0,0 +1,124 @@
+import asyncio
+import logging
+
+from ophyd import Component as Cpt
+from ophyd import DynamicDeviceComponent as DCpt
+from ophyd import Device, EpicsMotor, EpicsSignal
+
+from .._iconfig import load_config
+from .device import aload_devices, make_device
+
+log = logging.getLogger(__name__)
+
+def transfer_sample(num_dios: int):
+    """Create a dictionary with robot sample device definitions.
+    For use with an ophyd DynamicDeviceComponent.
+    Parameters
+    ==========
+    num_dios
+      How many samples to create.
+    """
+    sample = {}
+    sample_attrs = ['present', 'empty', 'load', 'unload','x', 'y', 'z', 'rx', 'ry', 'rz']
+    for n in range(num_dios):
+        for attr in sample_attrs:
+            sample[f"sample{n}_{attr}"] = Cpt(EpicsSignal, f":sample{n}:{attr}", labels={"transfer"})
+    return sample
+
+
+class Robot(Device):
+    # joints and position
+    i = Cpt(EpicsMotor, ":i", labels={ "joint"})
+    j = Cpt(EpicsMotor, ":j", labels={ "joint"})
+    k = Cpt(EpicsMotor, ":k", labels={ "joint"})
+    l = Cpt(EpicsMotor, ":l", labels={ "joint"})
+    m = Cpt(EpicsMotor, ":m", labels={ "joint"})
+    n = Cpt(EpicsMotor, ":n", labels={ "joint"})
+    x = Cpt(EpicsMotor, ":x", labels={ "joint"})
+    y = Cpt(EpicsMotor, ":y", labels={ "joint"})
+    z = Cpt(EpicsMotor, ":z", labels={ "joint"})
+    rx = Cpt(EpicsMotor, ":rx", labels={ "joint"})
+    ry = Cpt(EpicsMotor, ":ry", labels={ "joint"})
+    rz = Cpt(EpicsMotor, ":rz", labels={ "joint"})
+    acc = Cpt(EpicsMotor, ":acceleration", labels={ "joint"})
+    vel = Cpt(EpicsMotor, ":velocity", labels={ "joint"})
+    
+    # dashboard
+    remote_control = Cpt(EpicsSignal, ":dashboard:remote_control", labels={"dashboard"}, kind="config")
+    program_rbv = Cpt(EpicsSignal, ":dashboard:program_rbv", labels={"dashboard"}, kind="config")
+    installation = Cpt(EpicsSignal, ":dashboard:installation", labels={"dashboard"}, kind="config")
+    play = Cpt(EpicsSignal, ":dashboard:play", labels={"dashboard"}, kind="config")
+    stop = Cpt(EpicsSignal, ":dashboard:stop", labels={"dashboard"}, kind="config")
+    pause = Cpt(EpicsSignal, ":dashboard:pause", labels={"dashboard"}, kind="config")
+    quit = Cpt(EpicsSignal, ":dashboard:quit", labels={"dashboard"}, kind="config")
+    shutdown = Cpt(EpicsSignal, ":dashboard:shutdown", labels={"dashboard"}, kind="config")
+    release_brake = Cpt(EpicsSignal, ":dashboard:release_brake", labels={"dashboard"}, kind="config")
+    close_safety_popup = Cpt(EpicsSignal, ":dashboard:close_safety_popup", labels={"dashboard"}, kind="config")
+    unlock_protective_stop = Cpt(EpicsSignal, ":dashboard:unlock_protective_stop", labels={"dashboard"}, kind="config")
```

Comment:
PVs that are set to '1' to perform some actions should not be included in the scan since they provide no useful information (they'll just be 0). Set their ``kind="omitted"``.